enhancements/sig-architecture/158-addon-v1beta1/README.md

```diff
@@ -0,0 +1,327 @@
+
+	// name and namespace of add-on configuration.
+	ConfigReferent `json:",inline"`
+}
+
+// ConfigGroupResource represents the GroupResource of the add-on configuration
+type ConfigGroupResource struct {
+	// group of the add-on configuration.
+	// +optional
+	// +kubebuilder:default=""
+	Group string `json:"group"`
+
+	// resource of the add-on configuration.
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength:=1
+	Resource string `json:"resource"`
+}
+
+// ConfigReferent represents the namespace and name for an add-on configuration.
+type ConfigReferent struct {
+	// namespace of the add-on configuration.
+	// If this field is not set, the configuration is in the cluster scope.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// name of the add-on configuration.
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength:=1
+	Name string `json:"name"`
+}
+```
+
+#### Conversion Webhook Design
+- The conversion webhook will convert v1alpha1 `spec.supportedConfigs` to v1beta1 `spec.defaultConfigs`.
+- `PLACEHOLDER` as a **reserved name** will be used in `spec.defaultConfigs.name` to store the converted v1alpha1 API without default configs.
+
+For example:
+
+**Case 1: `ClusterManagementAddOn` `spec.supportedConfigs` is defined in v1alpha1**
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: application-manager
+spec:
+  supportedConfigs:
+  - group: addon.open-cluster-management.io
+    resource: addondeploymentconfigs
+---
+apiVersion: addon.open-cluster-management.io/v1beta1
+kind: ClusterManagementAddOn
+metadata:
+  name: application-manager
+spec:
+  defaultConfigs:
+  - group: addon.open-cluster-management.io
+    resource: addondeploymentconfigs
+    name: PLACEHOLDER
+```
+
+**Case 2: `spec.supportedConfigs.defaultConfig` is defined in v1alpha1**
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: application-manager
+spec:
+  supportedConfigs:
+  - defaultConfig:
+      name: default-deploy-config
+      namespace: open-cluster-management
+    group: addon.open-cluster-management.io
+    resource: addondeploymentconfigs
+---
+apiVersion: addon.open-cluster-management.io/v1beta1
+kind: ClusterManagementAddOn
+metadata:
+  name: application-manager
+spec:
+  defaultConfigs:
+  - group: addon.open-cluster-management.io
+    resource: addondeploymentconfigs
+    name: default-deploy-config
+    namespace: open-cluster-management
+```
+
+### ManagedClusterAddon registrations
+
+#### Key Change
+Redesign `status.registrations` to be more flexible for future addon registration patterns in v1beta1.
+
+#### API Definition & Conversion Webhook
+More details in [Addon registration enhancement](https://docs.google.com/document/d/1JO-GEhhMwJQeDFvApcC-JU63bnfWsMb-9QRX-gzhAqc/edit?tab=t.0#heading=h.k654poy68xzg).
```

Comment:
I think at the minimum we should setup the API shape here in v1beta1 as

```yaml
registration:
- type: kubeClient
   kubeClient:
      user:
        name: foo
        groups: ["group1"]
- type: csr
   csr:
     signerName: signer1
     subject:
       user: foo
       groups: ["groups1"]
       organizationUnits: ["group1"]
```

This should be good enough for this proposal and we can provide a conversion with this.

The addon registration driver can be enhanced based on this API